### PR TITLE
Add support for container signing and SBOMs

### PR DESCRIPTION
**Pull request by scholzj**
This PR adds support for container signing and SBOMs to the Access Operator CI and Makefiles.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,3 +109,42 @@ Learn more on how you can contribute on our [Join Us](https://strimzi.io/join-us
 ## License
 
 Strimzi Access Operator is licensed under the [Apache License](./LICENSE), Version 2.0
+
+## Container signatures
+
+Strimzi Access Operator containers are signed using the [`cosign` tool](https://github.com/sigstore/cosign).
+Strimzi currently does not use the keyless signing and the transparency log.
+To verify the container, you can copy the following public key into a file:
```
Comment (PaulRMellor) [inline README.md:117]:
```suggestion
To verify the authenticity of the container, you can copy the following Strimzi public key into a file:
```

File: README.md [hunk 2]
```diff
@@ -109,3 +109,42 @@ Learn more on how you can contribute on our [Join Us](https://strimzi.io/join-us
...
+And use it to verify the signature:
+
+```
+cosign verify --key strimzi.pub quay.io/strimzi/kafka-access-operator:latest --insecure-ignore-tlog=true
+```
+
+## Software Bill of Materials (SBOM)
+
+Strimzi Access Operator publishes the software bill of materials (SBOM) of our containers.
+The SBOMs are published as an archive with `SPDX-JSON` and `Syft-Table` formats signed using cosign.
```
Comment (PaulRMellor) [inline README.md:135]:
```suggestion
The SBOMs are published as archives in `SPDX-JSON` and `Syft-Table` formats, and they are  signed using cosign.
```

File: README.md [hunk 3]
```diff
@@ -109,3 +109,42 @@ Learn more on how you can contribute on our [Join Us](https://strimzi.io/join-us
...
+To verify the container, you can copy the following public key into a file:
+
+```
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET3OleLR7h0JqatY2KkECXhA9ZAkC
+TRnbE23Wb5AzJPnpevvQ1QUEQQ5h/I4GobB7/jkGfqYkt6Ct5WOU2cc6HQ==
+-----END PUBLIC KEY-----
+```
+
+And use it to verify the signature:
```
Comment (PaulRMellor) [inline README.md:126]:
```suggestion
Use the following cosign command to verify the signature:
```

File: README.md [hunk 4]
```diff
@@ -109,3 +109,42 @@ Learn more on how you can contribute on our [Join Us](https://strimzi.io/join-us
...
+```
+cosign verify --key strimzi.pub quay.io/strimzi/kafka-access-operator:latest --insecure-ignore-tlog=true
+```
+
+## Software Bill of Materials (SBOM)
+
+Strimzi Access Operator publishes the software bill of materials (SBOM) of our containers.
+The SBOMs are published as an archive with `SPDX-JSON` and `Syft-Table` formats signed using cosign.
+For releases, they are also pushed into the container registry.
+To verify the SBOM signatures, please use the Strimzi public key:
```
Comment (PaulRMellor) [inline README.md:137]:
```suggestion
To verify the authenticity of the SBOM signatures, use the Strimzi public key:
```

File: README.md [hunk 5]
```diff
@@ -109,3 +109,42 @@ Learn more on how you can contribute on our [Join Us](https://strimzi.io/join-us
...
+
+```
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET3OleLR7h0JqatY2KkECXhA9ZAkC
+TRnbE23Wb5AzJPnpevvQ1QUEQQ5h/I4GobB7/jkGfqYkt6Ct5WOU2cc6HQ==
+-----END PUBLIC KEY-----
+```
+
+You can use it to verify the signature of the SBOM files with the following command:
```
Comment (PaulRMellor) [inline README.md:146]:
```suggestion
Use the following cosign command to verify the signatures:
```

**Review by PaulRMellor — APPROVED** [review]
Looks good. I made a couple of suggestions for the readme

**Review by ppatierno — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**